test.js

```diff
@@ -0,0 +1,176 @@
+'use strict'
+
+const { test } = require('tap')
+const Fastify = require('fastify')
+const sget = require('simple-get').concat
+const cors = require('cors')
+
+const expressPlugin = require('./index')
+
+test('Should support connect style middlewares', t => {
+  t.plan(3)
+
+  const fastify = Fastify()
+
+  fastify
+    .register(expressPlugin)
+    .after(() => fastify.use(cors()))
+    .listen(0, (err, address) => {
+      t.error(err)
+      sget({
+        method: 'GET',
+        url: address
+      }, (err, res, data) => {
+        t.error(err)
+        t.match(res.headers, {
+          'access-control-allow-origin': '*'
+        })
+        fastify.close()
+      })
+    })
+})
+
+test('Middleware chain', t => {
+  t.plan(5)
+
+  const order = [1, 2, 3]
+  const fastify = Fastify()
+
+  fastify
+    .register(expressPlugin)
+    .after(() => {
+      fastify
+        .use(middleware1)
+        .use(middleware2)
+        .use(middleware3)
+    })
+    .listen(0, (err, address) => {
+      t.error(err)
+      sget({
+        method: 'GET',
+        url: address
+      }, (err, res, data) => {
+        t.error(err)
+        fastify.close()
+      })
+    })
+
+  function middleware1 (req, res, next) {
+    t.strictEqual(order.shift(), 1)
+    next()
+  }
+
+  function middleware2 (req, res, next) {
+    t.strictEqual(order.shift(), 2)
+    next()
+  }
+
+  function middleware3 (req, res, next) {
+    t.strictEqual(order.shift(), 3)
+    next()
+  }
+})
+
+test('Middleware chain (with errors)', t => {
```

Comment:
There should also be a test that tests what happens if an error occurs in a middleware and there are no "error-handling" middlewares. In that case, I would expect the error to end up in Fastify's error handler.